app/lib/pre_assembly/batch.rb

```diff
@@ -112,21 +115,17 @@ def process_digital_objects
         num_no_file_warnings += 1 if dobj.object_files.empty?
         progress = { dobj: dobj }
         file_attributes_supplied = batch_context.all_files_public? || dobj.dark?
-        begin
-          # Try to pre_assemble the digital object.
-          load_checksums(dobj)
-          status = dobj.pre_assemble(file_attributes_supplied)
-          # Indicate that we finished.
-          progress[:pre_assem_finished] = true
-          log "Completed #{dobj.druid}"
-        ensure
-          # Log the outcome no matter what.
-          File.open(progress_log_file, 'a') { |f| f.puts log_progress_info(progress, status || incomplete_status).to_yaml }
-        end
+        load_checksums(dobj)
+        progress = dobj.pre_assemble(file_attributes_supplied)
+        num_failures += 1 if progress[:status] == 'error'
+        log "Completed #{dobj.druid}"
+        File.open(progress_log_file, 'a') { |f| f.puts progress.merge(timestamp: Time.now.utc.strftime('%Y-%m-%d %H:%M:%S')).to_yaml }
       end
-    ensure
       log "**WARNING**: #{num_no_file_warnings} objects had no files" if num_no_file_warnings > 0
-      log "#{total_obj || 0} objects pre-assembled"
+      log "**WARNING**: #{num_failures} objects had errors during pre-assembly" if num_failures > 0
+      log "#{total_obj} objects pre-assembled"
+
+      num_failures == 0 # if no failures in any objects, return true, else false
     end
```

Comment:
if we have any errors during the processing of any single digital object, we will enumerate these, so this method can return true or false to the caller, allowing us to transition to the correct state for the overall job run